src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1261,185 +1201,64 @@ export class NetworkAreaDiagramViewer {
-            isTransformerEdge,
-            nodeRadius2,
-            edgeInfoId2,
-            bentLine
-        );
-
         if (isTransformerEdge) {
-            this.redrawTransformer(
-                edgeNode,
-                halfEdgePoints1.at(-2)!,
-                halfEdgePoints1.at(-1)!,
-                halfEdgePoints2.at(-2)!,
-                halfEdgePoints2.at(-1)!,
-                edgeType
-            );
+            this.redrawTransformer(edgeNode, halfEdge1, halfEdge2, edgeType);
         } else if (isHVDCLineEdge) {
-            this.redrawConverterStation(
-                edgeNode,
-                halfEdgePoints1.at(-2)!,
-                halfEdgePoints1.at(-1)!,
-                halfEdgePoints2.at(-2)!,
-                halfEdgePoints2.at(-1)!
-            );
+            this.redrawConverterStation(edgeNode, halfEdge1, halfEdge2);
         }
         // if present, move edge name
         if (this.svgParameters.getEdgeNameDisplayed()) {
-            this.updateEdgeName(edgeNode, halfEdgePoints1.at(-1)!, halfEdgePoints2.at(-2)!);
+            this.updateEdgeName(edgeNode, halfEdge1.edgePoints.at(-1)!, halfEdge2.edgePoints.at(-2)!);
         }
         // store edge angles, to use them for bus node redrawing
-        this.edgeAngles1.set(edgeNode.id, DiagramUtils.getAngle(halfEdgePoints1[0], halfEdgePoints1[1]));
-        this.edgeAngles2.set(edgeNode.id, DiagramUtils.getAngle(halfEdgePoints2[0], halfEdgePoints2[1]));
+        this.edgeAngles1.set(edgeNode.id, DiagramUtils.getEdgeStartAngle(halfEdge1));
+        this.edgeAngles2.set(edgeNode.id, DiagramUtils.getEdgeStartAngle(halfEdge2));
     }
 
-    private redrawHalfEdge(
-        edgeNode: SVGGraphicsElement,
-        side: string,
-        polylinePoints: Point[],
-        transformerEdge: boolean,
-        nodeRadius: [number, number, number],
-        edgeInfoId: string | undefined,
-        bentLine: boolean
-    ) {
+    private redrawHalfEdge(edgeNode: SVGGraphicsElement, halfEdge: HalfEdge) {
         // move edge polyline
         const polyline: SVGGraphicsElement | null = edgeNode.querySelector(
-            ':scope > polyline.nad-edge-path:nth-of-type(' + side + ')'
+            ':scope > polyline.nad-edge-path:nth-of-type(' + halfEdge.side + ')'
         );
-
-        // if transformer edge reduce edge polyline, leaving space for the transformer
-        if (transformerEdge) {
-            polylinePoints[polylinePoints.length - 1] = DiagramUtils.getPointAtDistance(
-                polylinePoints.at(-1)!,
-                polylinePoints.at(-2)!,
-                1.5 * this.svgParameters.getTransformerCircleRadius()
-            );
-        }
-
-        const polylinePointsFormatted: string = polylinePoints.map((p) => DiagramUtils.getFormattedPoint(p)).join(' ');
-        polyline?.setAttribute('points', polylinePointsFormatted);
+        const points: string = halfEdge.edgePoints.map((p) => DiagramUtils.getFormattedPoint(p)).join(' ');
```

Comment:
Here a getFormattedPolyline function, taking as input a list of points, could be used